src/Symfony/Component/Translation/Tests/PseudoLocalizationTranslatorTest.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Translation\Tests;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Translation\IdentityTranslator;
+use Symfony\Component\Translation\PseudoLocalizationTranslator;
+
+final class PseudoLocalizationTranslatorTest extends TestCase
+{
+    /**
+     * @dataProvider provideTrans
+     */
+    public function testTrans(string $expected, string $input, array $options = []): void
+    {
+        $this->assertSame($expected, (new PseudoLocalizationTranslator(new IdentityTranslator(), $options))->trans($input));
+    }
+
+    public function provideTrans(): array
+    {
+        return [
+            ['[ƒöö⭐ ≤þ≥ƁÅŔ≤⁄þ≥]', 'foo⭐ <p>BAR</p>'], // Test defaults
+            ['before <div data-label="fcy"><a href="#" title="bar" data-content="ccc">foo</a></div> after', 'before <div data-label="fcy"><a href="#" title="bar" data-content="ccc">foo</a></div> after', $this->getIsolatedOptions(['parse_html' => true])],
+            ['ƀéƒöŕé <div data-label="fcy"><a href="#" title="bar" data-content="ccc">ƒöö</a></div> åƒţéŕ', 'before <div data-label="fcy"><a href="#" title="bar" data-content="ccc">foo</a></div> after', $this->getIsolatedOptions(['parse_html' => true, 'accents' => true])],
```

Comment:
please add some HTML examples having escaped chars in attributes and in the text.